### PR TITLE
Refactoring to replace unittests with pytests

### PR DESCRIPTION
**Pull request by abidlabs**
I started going through our tests and converting them from unittest to pytest, when I noticed something wrong -- some of our tests were passing even though they should not have been. It seems that pytest was letting async tests automatically pass if they were written as part of a unittest class. 

For example, if you change the `TestBlocksMethods::test_dict_inputs_in_config` test in `test_blocks.py` to anything else, you'll find that it still passes:

```py
    async def test_dict_inputs_in_config(self):
        1/0
```

This is not the case if we change the tests to be pure pytests. So I went through and converted all of our tests to pytest. I noticed some other tests that should not have been passing, either due to the tests being incorrect or there being bugs in our codebase. Nothing too major, but here are the changes I made:

* Many tests in `test_components.py` passed in a *list* to `Interface.__call__` instead of the input parameters as separate arguments. This is not correct and the tests have been fixed. 
* In `test_components.py`, several tests used `.serialize()` in a way that is not supported or used in our library anymore. They've been rewritten or removed.
* Many of the components were not raising a `ValueError()` when passed an invalid argument for `type`. They now correctly do this. 

All of the above issues are now fixed in the library, and all of our tests have been converted to pytest format. 

Closes: #1784

### CODE REVIEW

**Comment by freddyaboulton** [comment]
@abidlabs I noticed on Friday that pytest emits some warnings along the lines of “coroutine test_….. was never awaited”. Which means those tests are never really checked 😬 

Confusingly, these tests have “@pytest.mark.asyncio”. 

I wonder if this PR will get rid of those will get rid of those warnings.

**Comment by abidlabs** [comment]
Exactly @freddyaboulton it seems that some async tests are not running at all! l elaborate in a follow up message

**Comment by abidlabs** [comment]
I've updated the original message with more details, and this is now ready for review

File: test/test_components.py [hunk 1]
```diff
@@ -1,14 +1,20 @@
+"""
+Tests for all of the componets defined on components.lpy. They are divided into two types of tests:
```
Comment (freddyaboulton) [inline test/test_components.py:2]: nit typo: `components.lpy`

**Review by freddyaboulton — APPROVED** [review]
Thanks for the fix @abidlabs ! I think the fact some components now raise ValueError about invalid `type` parameter values is worthy of a mention in the changelog though.

File: test/test_components.py (hunk 1 above)
Comment (abidlabs) [reply test/test_components.py:2]: Ah thanks

**Comment by abidlabs** [comment]
That's fair, thanks @freddyaboulton for the review! Will add a changelog and merge in to avoid conflicts

**Pull request merged**